File: test/Authzi.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common/Initialization/TestData.fs
namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common

open AuthZI.Deploy.MicrosoftEntra.Configuration
open AuthZI.Deploy.MicrosoftEntra.Configuration.Credentials.MicrosoftEntraID1
open AuthZI.Identity.MicrosoftEntra
open AuthZI.MicrosoftEntra
open Orleans

type public TestData() =
    static member val public UserWithScopeAdeleV : obj[] list = [] with get, set

    static member val public UserWithScopeAlexW : obj[] list = [] with get, set

    static member val public Users : obj[] list = [] with get, set

    static member val public AzureActiveDirectoryApp : MicrosoftEntraApp = 
        MicrosoftEntraIDApp.EmptyApp with get, set
    
    static member val public Web1Client : Client = 
        Credentials.MicrosoftEntraID1.WebClient1 with get, set

    static member val IClusterClient : IClusterClient = null with get, set

    static member val GetAccessTokenForUserOnWebClient1Async : 
        MicrosoftEntraApp -> string -> string -> string = fun app user pass -> "" with get, set

File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common.CSharp/Initialization/TestData.cs
using AuthZI.Identity.MicrosoftEntra;
using Orleans;
using System.Collections.Generic;

namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common.Initialization;

public class TestData
{
  public static IEnumerable<object[]> UserWithScopeAdeleV { get; set; } = [];

  public static IEnumerable<object[]> UserWithScopeAlexW { get; set; } = [];

  public static IEnumerable<object[]> Users { get; set; } = [];

  public static IReadOnlyDictionary<string, string> UserPasswords { get; set; } =
    new Dictionary<string, string>();

  public static MicrosoftEntraApp WebClient1 { get; set; } = MicrosoftEntraIDApp.EmptyApp;

  public static MicrosoftEntraApp WebClient2 { get; set; } = MicrosoftEntraIDApp.EmptyApp;

  public static IClusterClient IClusterClient { get; set; } = null!;
}
